kitsune/products/models.py

```diff
@@ -188,3 +188,71 @@ class Platform(ModelBase):
 
     def __str__(self):
         return "%s" % self.name
+
+
+class Banner(ModelBase):
+    uploaded_image = models.ImageField(
+        upload_to=settings.BANNER_IMAGE_PATH,
+        null=True,
+        blank=True,
+        max_length=settings.MAX_FILEPATH_LENGTH,
+        help_text="Must be 72x72px.",
+    )
+    static_image = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Path to an image in staticfiles, e.g. 'protocol/img/logos/firefox/logo.svg'. "
+        "Takes precedence over uploaded_image, leave both blank to show no image.",
+    )
+    description = models.TextField()
+    link = models.CharField(max_length=255, blank=True)
+    button = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Text to show on button, leave blank to show no button.",
+    )
+    modifer_class = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text="Additional classes to add to the banner, space seperated.",
+    )
+
+    products = models.ManyToManyField(
+        Product,
+        blank=True,
+        help_text="Products to associate the banner with. "
+        "Leave blank to associate with all products.",
+    )
+    topics = models.ManyToManyField(
+        Topic,
+        blank=True,
+        help_text="Topics to associate the banner with. Leave blank to associate with all topics.",
+    )
+
+    visible = models.BooleanField()
+    priority = models.IntegerField(
+        default=0, help_text="Higher priority banners will show instead of lower priority ones."
+    )
+
+    show_to_authenticated = models.BooleanField(
+        default=True,
+        verbose_name="Show to logged in users",
+    )
+    show_to_unauthenticated = models.BooleanField(
+        default=True,
+        verbose_name="Show to logged out users",
+    )
+
+    show_on_topic = models.BooleanField(
+        verbose_name="Show on article listing (topic) pages",
+        help_text="Will only show if the banner is associated "
+        "with the topic and product of the topic page.",
+    )
+    show_on_article = models.BooleanField(
+        verbose_name="Show on articles",
+        help_text="Will only show if the banner is associated "
```

Comment:
the description is a bit confusing. I guess this flag toggles whether it will be displayed in the articles under the specified product and or topics if set?

Reply:
Functionality wise I would approach it as follows:

* If a product is set -> Display under that product only.
* If a topic is set -> Display under the product and the set of selected topics
* If the flag is set to display in KB -> Display in all articles under the product if not topic is set, otherwise display to all articles under the topic set

Reply:
This toggles whether it appears in the article itself (a la the existing FxA banner), the `show_on_article` bool toggles if it appears on a topic page (e.g. this page: https://support.mozilla.org/en-US/products/firefox/privacy-and-security). You may want it to display on either, or both - and this system of logic allows adding other `show_on_` conditionals if we need them (e.g. `show_on_landing`, `show_on_aaq`, etc.)